src/sync/app.cpp

```diff
@@ -36,16 +36,36 @@ Optional<T> get_optional(const nlohmann::json& json, const std::string& key)
     return it != json.end() ? Optional<T>(it->get<T>()) : realm::util::none;
 }
 
-static std::map<std::string, std::string> get_request_headers(std::shared_ptr<SyncUser> with_user_authorization = nullptr)
+enum class RequestTokenType {
+    NoAuth,
+    AccessToken,
+    RefreshToken
+};
+
+// generate the request headers for a HTTP call, by default it will generate headers with a refresh token if a user is passed
+static std::map<std::string, std::string> get_request_headers(std::shared_ptr<SyncUser> with_user_authorization = nullptr,
```

Comment:
You can also remove `with_user_authorization`. It seems to be covered by the `NoAuth` case?

Reply:
The purpose of that is so you can specify the user, as opposed to just relying on the `current_user` method

Reply:
Ah yeah, sorry. Missed you got the access/refresh token from it.